capnpc-go/templates/promiseFieldAnyList

```diff
@@ -0,0 +1,3 @@
+func (p {{ .Node.Name }}_Future) {{ .Field.Name|title }}() *capnp.Future {
```

Comment:
I feel like we should define and return a List_Future type, rather than continuing to just use plain Future; that's sortof the point of specializing AnyList in the first place, no?

Reply:
As per discussion on Matrix, I am merging now and we can add this in a separate PR.